New homepage architecture

clementbiron opened the pull request.

MattiSG commented on src/modules/Common/containers/Container.module.css:
```diff
@@ -92,10 +92,10 @@
     background-color: var(--colorBlack200);
-
+    /* 
     & hr {
       border-color: var(--colorWhite);
-    }
+    } */
```
Please erase code if it is not used instead of commenting it out, otherwise we don't know what is useful and what is not 🙂

MattiSG commented on src/modules/Common/styles/elements/utility.css:
```diff
@@ -18,6 +18,12 @@
   text-align: left;
 }
 
+/* Margin center */
```
Not sure this comment is so useful 😉

MattiSG commented on src/modules/Contribute/pages/home.tsx:
```diff
@@ -51,25 +40,19 @@ const ContributeHomePage = () => {
-                  name: 'Contribute',
-                  url: './#' + t('contribute/home:contribute.id', 'contribute'),
+                  name: t('contribute:breadcrumb.home_page.name'),
+                  url: 'https://www.opentermsarchive.org',
```
```suggestion
                  url: 'https://opentermsarchive.org',
```

MattiSG commented on src/modules/OTA-api/data-components/Contributors.tsx:
```diff
@@ -1,91 +1,189 @@
+    login: 'Matti Schneider',
+    type: 'core',
+    avatar_url: 'https://avatars.githubusercontent.com/u/222463?v=4',
+    html_url: 'https://github.com/MattiSG',
```
```suggestion
    html_url: 'https://mattischneider.fr',
```

MattiSG commented on src/modules/OTA-api/data-components/Contributors.tsx:
```diff
@@ -1,91 +1,189 @@
+    html_url: 'https://github.com/karnauskas',
+  },
+  {
+    login: 'TOSDR',
+    type: 'contributors',
+    avatar_url: 'https://avatars.githubusercontent.com/u/3402969?s=200&v=4',
+    html_url: 'https://github.com/tosdr',
```
This is not a contributor, this is a bot.

clementbiron replied:
The `type` here is not related to the GitHub account type. 
The following param values `core` `alumnis` `contributors` `all` allow to display the `Contributors` component with associated data.

MattiSG commented on src/modules/OTA-api/data-components/Contributors.tsx:
```diff
@@ -1,91 +1,189 @@
+    html_url: 'https://github.com/SiegridHenry',
+  },
+  {
+    login: 'Chirsitan Quest',
```
```suggestion
    login: 'Christian Quest',
```

MattiSG commented on src/modules/OTA-api/data-components/Contributors.tsx:
```diff
@@ -1,91 +1,189 @@
-const Contributors: React.FC<ContributorsProps> = React.memo(({ subtitle }) => {
-  const { data } = useSWR('/api/ota/contributors/all', {
-    revalidateOnMount: true,
+const getContributorsByType = (type: ContributorsProps['type']) => {
```
I think it would be simpler and more efficient to maintain three different arrays of contributors rather than adding a property to all contributors and then loop over them three times on every page load 😉

MattiSG commented on src/translations/default/about.mdx:
```diff
@@ -0,0 +1,45 @@
+**CONTENT_NOT_TRANSLATED**
```
Is this where the English translation is supposed to happen? Should this stay this way?

clementbiron replied:
As discussed every english translations are located in `translations/en/` folder

martinratinaud replied:
This default folder is due to a know problem on `next-i18next`. This is really bad and should be removed

The only way I found was to switch to next-translate which is also a lot easier to use.

We can try that in a next iteration

MattiSG commented on src/translations/default/about.mdx:
```diff
@@ -0,0 +1,45 @@
+##### Financeurs
+
+<ThumbGalery align="left" small={true}>
+  <ThumbGaleryItem src="/images/logo-meae.png" width="132" height="120" />
+  <ThumbGaleryItem src="/images/france-relance.png" width="77" height="77" />
+  <ThumbGaleryItem src="/images/nextgen.png" width="191" height="48" />
```
Is this duplicated in every language then?! This is going to be a nightmare to maintain.

clementbiron replied:
I don't have a better solution for now and even if we have to change the page 3 times a year in 5 languages it doesn't seem so much a nightmare to me.

MattiSG reviewed (changes requested):
Thank you for these changes. The resulting homepage is much more focused and looks clean! 👍

On the code side, I don't believe I can accurately review it. The complexity of the stack, combined with a lack of clarity around what is generated, included from a third-party framework, or written manually, makes it too hard to review this massive changeset. I am also surprised by a lot of diff that seems like it should have been handled in #55, #56 and #57.

Beyond line-level comments, here are comments on file names:

These files are not up to date:
- `android-chrome-72x72.png`
- `android-chrome-96x96.png`

These files are added while they are not used:
- `public/images/detection.svg`
- `public/images/diff.svg`
- `public/images/diff.png`
- `public/images/logo/logo-open-terms-archive-black-alternative.png`
- `public/images/logo/logo-open-terms-archive-black-alternative.svg`
…and a few others.

Misnamed:

- `images/unknow.png` (should be `unknown.png`, and preferably `new-reuse.png` for clearer usage).

Please use ImageOptim to decrease the file size of pictures prior to committing.

I believe the images would be easier to maintain if they were stored in folders according to their type. For example, a subfolder called `reuses` for the screenshots, another one called `contributors` for the logos, `how-it-works` for the 4 steps…

MattiSG commented:
Please include @avernois in the list of contributors! 🙂

clementbiron commented:
Thanks Matti.

- I cleaned unused images, run image optim, use subfolder for better discoverability and rename misnamed files ✅
- Added @avernois to the contributor list ✅
- I change www.opentermarchive.org to opentermsarchive.org ✅
- Change url of Matti and Clement in contributor list  ✅
- Fix wording in `Christian` name ✅
- Use different arrays for contributors type ✅

MattiSG commented on src/pages/homepage.tsx:
```diff
@@ -1,482 +1,212 @@
+        <CardList title={t('homepage:reuses.title')} subtitle={t('homepage:reuses.subtitle')}>
+          <Card
+            image="/images/reuses/tosdr.jpg"
+            title="Terms of Service; Didn’t Read"
+            subtitle="Permet aux utilisateurs de donner collaborativement une note aux conditions d’utilisation selon le respect de leurs droits."
+            author="ToS;DR"
+            link="https://tosdr.org"
```
This is hardcoded, it should be loaded from translation 😉

clementbiron replied:
there was a time when I had glasses and it was not so bad, fixed 😅

MattiSG commented on src/pages/homepage.tsx:
```diff
@@ -1,482 +1,212 @@
+            image="/images/reuses/scripta-manent.jpg"
+            title={t('homepage:reuses.card1.title')}
+            subtitle={t('homepage:reuses.card1.subtitle')}
+            author={t('homepage:reuses.card1.author')}
```
Please use semantic names to support translation and maintenance, with same names between image file and translation keys 🙂 
```suggestion
            author={t('homepage:reuses.scripta-manent.author')}
```

MattiSG commented on src/pages/homepage.tsx:
```diff
@@ -1,482 +1,212 @@
+          <Card
+            image="/images/reuses/tosback.jpg"
+            title="TOSback"
+            subtitle="L’ancêtre d’Open Terms Archive ! Depuis octobre 2020, les changements affichés sont fournis par une instance dédiée d’Open Terms Archive."
```
Hardcoded 😉

MattiSG commented:
The dropped pages (How it works…) should be removed from the codebase.

MattiSG commented:
🤔 😅 

<img width="621" alt="Capture d’écran 2021-12-14 à 13 09 37" src="https://user-images.githubusercontent.com/222463/146035533-97c19223-218c-485e-91c7-c862ee834d2e.png">

MattiSG commented:
Too much “shuffle” maybe…

MattiSG commented:
The square version of the logo should be 2.7.1 from #63

clementbiron commented:
> The square version of the logo should be 2.7.1 from #63

There must be an error of understanding, for me the vertical logo that I put is the good one.

MattiSG reviewed (approved):
- I checked all links and corrected those that were wrong.
- I checked all translations, there is no `__STRING_NOT_TRANSLATED__` apart from in the `default` folder.
- I would like to have [version 2.7.1](https://github.com/ambanum/OpenTermsArchive.org/issues/63#issuecomment-992320359) as the square logo, but that's okay if it's not done now.
- I did not review most of the code itself.

martinratinaud commented on src/modules/Common/components/Divider.tsx:
```diff
@@ -0,0 +1,18 @@
+import s from './Divider.module.css';
+
+type DividerProps = {
+  className?: string;
```
no need for className

martinratinaud commented on src/modules/Common/components/Hero.tsx:
```diff
@@ -13,9 +13,8 @@ const Hero: React.FC<HeroProps> = ({ children, className, title, subtitle, ...pr
       {title ? <h1 className={s.hero_title}>{title}</h1> : undefined}
       {subtitle ? (
-        <div className={classNames(s.hero_subtitle, 'text__subtitle')}>{subtitle}</div>
+        <h2 className={classNames(s.hero_subtitle, 'h2__thin')}>{subtitle}</h2>
```
use `&&`

martinratinaud commented on src/modules/Common/components/Logo.tsx:
```diff
@@ -8,27 +10,27 @@ type LogoProps = {
   className?: string;
```
no need for `className`

martinratinaud commented on src/modules/Common/components/SubscribeForm.tsx:
```diff
@@ -97,84 +91,51 @@ const SubscribeForm = ({
-                'We take care of your data and do not sell it, check our privacy policy.'
-              )}
-            </p>
+            <p className="text__light">{t('common:subscribe_form.p3')}</p>
```
weird that p3 p1 and p3, maybe a mistake

martinratinaud commented on src/modules/Common/components/ThumbGalery.tsx:
```diff
@@ -7,6 +7,7 @@ type ThumbGaleryProps = {
   subtitle?: string;
```
ThumbGalery should be ThumbGallery (2 l)

martinratinaud commented on src/modules/Common/containers/CommonHead.tsx:
```diff
@@ -0,0 +1,69 @@
+
+      <link rel="alternate" href={`${currentUrl}`} hrefLang="x-default" />
+      {(router?.locales || [])
+        .filter((locale) => locale !== '__default')
```
We use `default` and not `__default`

martinratinaud commented on src/modules/OTA-api/data-components/Contributors.tsx:
```diff
@@ -1,91 +1,187 @@
+  subtitle?: string;
+  type?: 'core' | 'alumnis' | 'contributors' | 'all';
+  alignX?: 'left' | 'center' | 'right';
+  className?: string;
```
no need for className

martinratinaud commented on src/pages/about.tsx:
```diff
@@ -0,0 +1,43 @@
+            components={{
+              Contributors: Contributors,
+              Link: Link,
+              ThumbGalery: ThumbGalery,
```
You can use shorthand

```
components={{
Contributors,
Link,
ThumbGalery,
ThumbGaleryItem,
}}
```

and you also miss one `l`

martinratinaud commented on src/pages/case-studies.tsx:
```diff
@@ -13,26 +14,24 @@ import { useTranslation } from 'next-i18next';
       <Container gridCols="9" gridGutters="8">
         <TextContent>
-          <MDXRemote {...(mdxContent as any)} components={{ Button: Button }} />
+          <MDXRemote {...(mdxContent as any)} components={{ LinkIcon: LinkIcon }} />
```
use shorthand `components={{ LinkIcon }}`

clementbiron commented on src/pages/homepage.tsx:
```diff
@@ -96,47 +96,51 @@ const HomePage = () => {
           <ButtonBlockList title={t('homepage:cta_public.title')}>
             <ButtonBlock
-              title={t('homepage:cta_public.bloc1.title')}
+              title={t('homepage:cta_public.case_studies.title')}
```
In the case of these blocks, to allow the text to evolve without changing the key, I would have kept a generic name.

martinratinaud commented on src/translations/default/media.mdx:
```diff
@@ -0,0 +1,116 @@
+<a
+  target="_blank"
+  className="float__right"
+  https:title="An article to be found on the website of Wired" //www.wired.com/story/nations-need-ambassadors-to-big-tech/"
```
href missing

martinratinaud replied:
shouuld be `rel="noopener"`

martinratinaud commented on src/translations/en/about.mdx:
```diff
@@ -0,0 +1,44 @@
+
+#### Funders
+
+<ThumbGalery align="left" small={true} className="mb__XL">
```
2 l

martinratinaud reviewed (approved):
All reviewed

Well done, looks good

Some minor React tweaks to use

👏

clementbiron commented on src/modules/Common/components/ButtonBlock.tsx:
```diff
@@ -43,7 +43,7 @@ const ButtonBloc: React.FC<ButtonBlocProps> = ({
       {...props}
     >
       {icon ? <div className={classNames(s.buttonBlock_icon)}>{icon}</div> : undefined}
```
thanks !

clementbiron commented on src/modules/Common/components/Article.tsx:
```diff
@@ -20,12 +20,12 @@ const Article: React.FC<ArticleProps> = ({
-        {subtitle ? (
-          <div className={classNames('text__smallcaps', s.article_subtitle)}>{subtitle}</div>
-        ) : undefined}
         {title && titleLevel === 'h1' ? <h1 className={s.article_title}>{title}</h1> : undefined}
```
I modified it by using the same tricks as in `ThumbGalery.tsx` file :

```
const TitleComponent = titleLevel;
{title && <TitleComponent className={s.article_title}>{title}</TitleComponent>}
```

clementbiron commented on src/modules/Common/components/ButtonBlockList.tsx:
```diff
@@ -18,7 +18,11 @@ const ButtonBlocList: React.FC<ButtonBlocListProps> = ({
     <div className={classNames(s.buttonBlockList, className)} {...props}>
       <div className={classNames(s.buttonBlockList_header)}>
-        <div className={classNames(s.buttonBlockList_subtitle, 'text__smallcaps')}>{subTitle}</div>
+        {subTitle ? (
```
> `subTitle` or `subtitle` -> typescript must have yelled on that, didn't it?

Nop !

> and same `{subtitle && ...}

Done
